### PR TITLE
- Add a doom that will not expire until it does not matter anymore.

### PR DESCRIPTION
**Pull request by baldersheim**
- Doom can never be null.
- Wire doom into fillBitVector. Using it will be a separate PR.

@havardpe or @geirst PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/postinglistsearchcontext.hpp [hunk 1]
```diff
@@ -95,6 +99,7 @@ struct PostingListSearchContextT<DataT>::FillPart : public vespalib::Runnable {
             if (__builtin_expect(key < _docIdLimit, true)) { _bv->setBit(key); }
         });
     }
+    const vespalib::Doom       _doom;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/postinglistsearchcontext.hpp:102]: Was it intentional to take a copy here?

**Review by geirst — DISMISSED** [review]
👍

File: vespalib/src/vespa/vespalib/util/doom.h [hunk 2]
```diff
@@ -19,6 +19,7 @@ class Doom {
     duration soft_left() const noexcept { return _softDoom - _clock.getTimeNS(); }
     duration hard_left() const noexcept { return _hardDoom - _clock.getTimeNS(); }
     bool isExplicitSoftDoom() const noexcept { return _isExplicitSoftDoom; }
+    static const Doom & armageddon() noexcept;
```
Comment (havardpe) [inline vespalib/src/vespa/vespalib/util/doom.h:22]: what about 'never' ?

**Review by havardpe — DISMISSED** [review]

File: searchlib/src/vespa/searchlib/attribute/postinglistsearchcontext.hpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/attribute/postinglistsearchcontext.hpp:102]: Yes, you avoid an indirection as it wraps a reference.

File: vespalib/src/vespa/vespalib/util/doom.h (hunk 2 above)
Comment (baldersheim) [reply vespalib/src/vespa/vespalib/util/doom.h]: Fixed, and simplified as discussed.

**Pull request merged**